### PR TITLE
Maintenance incident fix

### PR DESCRIPTION
**Pull request by adrianschroeter**
yes, I know the test case is missing, try to do that tomorrow.

### CODE REVIEW

**Comment by bgeuken** [comment]
LGTM, from what I can tell. A test would be nice:-)

File: src/api/app/models/bs_request_permission_check.rb [hunk 1]
```diff
@@ -52,9 +52,8 @@ def check_accepted_action(action)
       begin
         ActiveXML.backend.direct_http(url)
       rescue ActiveXML::Transport::Error
-        # rubocop:disable Metrics/LineLength
-        raise ExpandError.new "The source of package #{action.source_project}/#{action.source_package}#{action.source_rev ? " for revision #{action.source_rev}" : ''} is broken"
-        # rubocop:enable Metrics/LineLength
+        raise ExpandError.new "The source of package #{action.source_project}/#{action.source_package}" +
```
Comment (ChrisBr) [inline src/api/app/models/bs_request_permission_check.rb]: You can make it even short when you remove the ``.new``

```
ExpandError, "The source of package #{action.source_project}/#{action.source_package}" +
```

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -851,6 +851,13 @@ def open_requests_with_by_package_review
     return BsRequest.where(id: rel.pluck('bs_requests.id'))
   end
 
+  def self.extended_name(project, package)
+    # the package name which will be used on a branch with extended or maintenance option
+    li = Directory.hashed(project: project, package: package)["linkinfo"]
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: Could you please name the variable ``linkinfo``?

File: src/api/app/models/package.rb [hunk 3]
```diff
@@ -851,6 +851,13 @@ def open_requests_with_by_package_review
...
+    return package + "." + li["project"] if li && li["project"]
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: What about checking first for project and using string interpolation?
```
project = li["project"] || project
"#{package}.#{project}"
```

Comment (adrianschroeter) [inline src/api/app/models/package.rb]: it would crash when li is nil

File: src/api/app/models/package.rb (hunk 2 above)
Comment (adrianschroeter) [inline src/api/app/models/package.rb]: it would make the next line too long IMHO

File: src/api/app/models/bs_request_permission_check.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/app/models/bs_request_permission_check.rb]: k, done in the new commit below.

File: src/api/app/models/package.rb (hunk 3 above)
Comment (ChrisBr) [inline src/api/app/models/package.rb]: What about?
```
project = li.try(:[],'project') || project
"#{package}.#{project}"
```

File: src/api/app/models/package.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/app/models/package.rb]: And with my suggestion for the next line?

File: src/api/app/models/bs_request_permission_check.rb [hunk 4]
```diff
@@ -39,7 +39,7 @@ class ReviewNotSpecified < APIException;
 
   def check_accepted_action(action)
     unless @target_project
-      raise NotExistingTarget.new "Unable to process project #{action.target_project}; it does not exist."
+      raise NotExistingTarget, "Unable to process project #{action.target_project}; it does not exist."
```
Comment (ChrisBr) [inline src/api/app/models/bs_request_permission_check.rb]: Can you rephrase your commit message?
```
 [api][webui] minor cleanup 
```
is not really self-explanatory ... (Maybe just use the description?)

**Comment by adrianschroeter** [comment]
urgs, catched the wrong branch, the fix is lost now :/

try to find it in local git again ...

**Comment by adrianschroeter** [comment]
restored the old version from production. The other changes are lost for now :/

**Comment by hennevogel** [comment]
What is the status of this thing now?

**Comment by adrianschroeter** [comment]
still technical correct and urgent to merge.

**Comment by bgeuken** [comment]
Then let's re-trigger the spider test run again. You should keep track of your PRs. Especially when they are urgent;-)

File: src/api/app/models/bs_request_permission_check.rb [hunk 5]
```diff
@@ -201,7 +207,8 @@ def set_permissions_for_action(action)
     # general write permission check on the target on accept
     @write_permission_in_this_action = false
     # meta data change shall also be allowed after freezing a project using force:
-    ignoreLock = opts[:force] && action.action_type == :set_bugowner
+    ignoreLock = (new_state == "declined")
+    ignoreLock = true if opts[:force] && action.action_type == :set_bugowner
```
Comment (bgeuken) [inline src/api/app/models/bs_request_permission_check.rb]: ``` ignoreLock = opts[:force] && action.action_type == :set_bugowner || new_state == "declined" ```

File: src/api/app/models/package.rb (hunk 3 above)
Comment (bgeuken) [inline src/api/app/models/package.rb]: @adrianschroeter Please update as requested:-)

Comment (adrianschroeter) [inline src/api/app/models/package.rb]: IMHO this is butt ugly and hard to read.... but if you want ...

File: src/api/app/models/package.rb (hunk 2 above)
Comment (bgeuken) [inline src/api/app/models/package.rb]: @adrianschroeter Could you change this one as well?

**Comment by bgeuken** [comment]
I've updated the commit as requested -> #2463
